src/main/java/capstone/mfslbackend/service/TransactionService.java

```diff
@@ -55,9 +62,56 @@ public Transaction createTransaction(Long fantasyTeamId, Long incomingPlayerId,
         }
 
         Player playerIn = playerService.getPlayerById(incomingPlayerId);
-        transaction.setPlayerIn(playerIn);
+        Player outgoingPlayer = playerService.getPlayerById(outgoingPlayerId);
+        String position = playerIn.getPosition();
+
+        if (playerIn.getTeam() == null) { //it means it is a free agent
 
-        transaction.setStatus(TransactionStatus.PROPOSED);
+            if (outgoingPlayer.getPosition().equals("Goalkeeper") && !(position.equals("Goalkeeper"))) {
+                    transaction.setStatus(TransactionStatus.REJECTED);
+                    throw new Error400("Fantasy team with id " + fantasyTeamId + " needs 1 goalkeeper");
+            } else if (position.equals("Goalkeeper")) {
+                if (proposingFantasyTeam.getPlayers().stream().filter(player -> player.getPosition().equals("Goalkeeper")).count() == MIN_GK) {
+                    transaction.setStatus(TransactionStatus.REJECTED);
+                    throw new Error400("Fantasy team with id " + fantasyTeamId + " already has a goalkeeper");
+                }
+            } else if (position.equals("Defender")) {
+                if (proposingFantasyTeam.getPlayers().stream().filter(player -> player.getPosition().equals("Defender")).count() == MAX_DEF && !outgoingPlayer.getPosition().equals("Defender")) {
```

Comment:
A fantasy team is allowed to have 9 Defenders if they want to have a bench full of defenders, but they're only allowed to have 5 in the first 11 of the playerIdsInOrder